File: 492A-Vanya and Cubes.py
def max_pyramid_height(n):
    height = 0
    cubes_used = 0
    i = 1
    
    while True:
        needed = i * (i + 1) // 2  # cubes required for level i
        if cubes_used + needed <= n:
            cubes_used += needed
            height += 1
            i += 1
        else:
            break
    return height


# Input
n = int(input())
print(max_pyramid_height(n))

File: 492A - Vanya and Cubes.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);

    int height = 0;
    int cubes_used = 0;
    int i = 1;

    while (1) {
        int needed = i * (i + 1) / 2;  // cubes needed for level i
        if (cubes_used + needed <= n) {
            cubes_used += needed;
            height++;
            i++;
        } else {
            break;
        }
    }

    printf("%d\n", height);
    return 0;
}

